### PR TITLE
SOC-601 | Set cookie on successful account registration

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-601

### CODE REVIEW

File: front/scripts/auth/signup/SignupForm.ts [hunk 1]
```diff
@@ -66,6 +65,8 @@ class SignupForm {
 			label: 'successful-registration'
 		});
 
+		document.cookie = 'registerSuccess=1; path=/; domain=' + this.getWikiaDomain();
```
Comment (garthwebb) [inline front/scripts/auth/signup/SignupForm.ts]: Would Cookie.set be good to add to ```front/scripts/auth/common/Cookie.ts``` ?

**Comment by garthwebb** [comment]
LGTM

File: front/scripts/auth/common/Cookie.ts [hunk 2]
```diff
@@ -1,12 +1,19 @@
+interface CookieAttributes {
+	expires?: string;
+	path?: string;
+	domain?: string;
```
Comment (bkoval) [inline front/scripts/auth/common/Cookie.ts]: AFAIK we are supposed to start introducing alphabetical order to interface properties.

**Comment by bkoval** [comment]
LGTM :palm_tree:

File: front/scripts/auth/signup/SignupForm.ts [hunk 3]
```diff
@@ -49,13 +49,12 @@ class SignupForm {
 		};
 	}
 
-	private trackValidationErrors(errors: Array<string>): void {
```
Comment (rafalkalinski) [inline front/scripts/auth/signup/SignupForm.ts]: Does it should be removed?

Comment (kvas-damian) [inline front/scripts/auth/signup/SignupForm.ts]: It's already present in: https://github.com/Wikia/mercury/blob/SOC-601/front/scripts/auth/common/FormErrors.ts#L73

File: front/scripts/auth/signup/SignupForm.ts [hunk 4]
```diff
@@ -66,6 +65,14 @@ class SignupForm {
 			label: 'successful-registration'
 		});
 
+		Cookie.set(
+			'registerSuccess',
+			'1',
+			{
+				domain: this.getWikiaDomain()
```
Comment (rafalkalinski) [inline front/scripts/auth/signup/SignupForm.ts]: if there will not be a redirect parameter (leading to the same origin), then user is redirected to the www.wikia.com, and then the cookie will not be accessible there

**Comment by rafalkalinski** [comment]
LGTM

**Pull request merged**